orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -447,7 +447,7 @@ Status OrtLoadOptimizerStatesInternal(const PathString& optimizer_folder_path,
         ParameterOptimizerState param_state;
         param_optimizer_states.insert({param_name, param_state});
       }
-      param_optimizer_states[param_name].momentum_named_states.insert({momentum_name, std::make_shared<OrtValue>(pair.second)});
+      param_optimizer_states[param_name].momentum_named_states.insert({momentum_name, std::move(pair.second)});
```

Comment:
optional: can use emplace for argument forwarding (since we are trying to avoid making copies).